File: spark-train/src/main/scala/com/sparktrain/scala/WordCount.scala
package com.sparktrain.scala

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.rdd.RDD

object WordCount {
  def main(args:Array[String]):Unit = {
    
    val inputFile = args(0)
      val outputFile = args(1)
    
    println("inputFile="+inputFile);
    
    println("outputFile="+outputFile);
    val conf = new SparkConf().setAppName("wordCount")
    
    val sc = new SparkContext(conf)
    //load our input data
    val input = sc.textFile(inputFile)
    
    println("input.count="+input.count())
    
    //split it up into words
   val words = input.flatMap(line => line.split(" "))
   println("words.count="+words.count());
   
    //transform into pairs and count.
    
    val counts = words.map(word => (word, 1)).reduceByKey{case (x, y) => x + y}
    
     println("counts.count="+counts.count());
    
    counts.saveAsTextFile(outputFile)
    
    
  }
}

File: spark-train/src/main/java/com/sparktrain/java/WordCount.java
package com.sparktrain.java;

import java.util.Arrays;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.FlatMapFunction;
import org.apache.spark.api.java.function.Function2;
import org.apache.spark.api.java.function.PairFunction;

import scala.Tuple2;

public class WordCount {

	public static void main(String[] args) {

		String inputFile = args[0];
		String outputFile = args[1];

		// Create a java spark context
		SparkConf conf = new SparkConf().setAppName("wordCount");
		JavaSparkContext sc = new JavaSparkContext(conf);
		// load our input data
		JavaRDD<String> input = sc.textFile(inputFile);
		// Split up into words
		JavaRDD<String> words = input
				.flatMap(new FlatMapFunction<String, String>() {
					public Iterable<String> call(String x) throws Exception {
						return Arrays.asList(x.split(" "));
					}
				});

		JavaPairRDD<String, Integer> counts = words.mapToPair(
				new PairFunction<String, String, Integer>() {
					public Tuple2<String, Integer> call(String x) {
						return new Tuple2(x, 1);
					}
				}).reduceByKey(new Function2<Integer, Integer, Integer>() {
			public Integer call(Integer x, Integer y) {
				return x + y;
			}
		});
		
		counts.saveAsTextFile(outputFile);
		
		JavaRDD<String> lines = sc.parallelize(Arrays.asList("pandas","i lik pandas"));
	}

}
